### PR TITLE
Feature/generalizing methods dag

### PR DESCRIPTION
**Pull request by AlejandroMus**

### CODE REVIEW

**Review by ggsdc — COMMENTED** [review]
Will review it after the other PR is merged

**Review by ggsdc — CHANGES_REQUESTED** [review]
You should add some tests to these methods

File: libs/client/cornflow_client/core/instance_solution.py [hunk 1]
```diff
@@ -34,6 +35,9 @@ def data(self) -> dict:
     def data(self, value: dict):
         self._data = value
 
+    def _get_property(self, key, prop) -> SuperDict:
+        return self.data[key].get_property(prop)
+
```
Comment (ggsdc) [inline libs/client/cornflow_client/core/instance_solution.py:40]: This method is not getting tested

File: libs/client/cornflow_client/core/instance_solution.py [hunk 2]
```diff
@@ -157,3 +161,252 @@ def to_excel(self, path: str):
...
+    @staticmethod
+    def get_hour_from_datetime_string(self, string: str) -> float:
+        """Returns the integer value of the hour (in number) from ts string in format 'YYYY-MM-DDTh:m'"""
+        return self.get_hour_from_ts(self.get_datetime_from_string(string))
+
+    @staticmethod
+    def get_hour_from_datetimesec_string(self, string: str) -> float:
+        """Returns the integer value of the hour (in number) from ts string in format 'YYYY-MM-DDTh:m:s'"""
+        return self.get_hour_from_ts(self.get_datetimesec_from_string(string))
```
Comment (ggsdc) [inline libs/client/cornflow_client/core/instance_solution.py:412]: These methods are not getting tested

**Review by ggsdc — CHANGES_REQUESTED** [review]

**Comment by ggsdc** [comment]
Just as a good practice. All these commits where pushed together to the repo: 

image

That can be seen because just the last one is the one that launched the automated workflows. When working on your local machine, you can ammend the commit and add more changes if the commit has not been pushed to the remote, that way all those commits could be just one.

**Review by ggsdc — DISMISSED** [review]

**Comment by ggsdc** [comment]
Will wait for the other PR to be merged before merging this one

**Review by marioncottard — DISMISSED** [review]

**Review by ggsdc — APPROVED** [review]

**Pull request merged**